contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -0,0 +1,147 @@
+use axelar_wasm_std::msg_id::HexTxHashAndEventIndex;
+use axelar_wasm_std::nonempty;
+use cosmwasm_std::{
+    Addr, Api, Event, HexBinary, QuerierWrapper, Response, Storage, Uint256, WasmMsg,
+};
+use error_stack::{report, Result, ResultExt};
+use router_api::client::Router;
+use router_api::{Address, ChainName, CrossChainId, Message};
+use sha3::{Digest, Keccak256};
+
+use crate::contract::Error;
+use crate::events::AxelarnetGatewayEvent;
+use crate::executable::AxelarExecutableClient;
+use crate::state::{self};
+
+#[allow(clippy::too_many_arguments)]
```

Comment:
should probably find a way to avoid having to use this. Maybe just pass in the config?

Reply:
If we add more fields to the config, then we're potentially moving more data around than needed. And deviates from the pattern of only passing data that's needed by the method, which is a good one to follow.

Reply:
I don't think you need to add more fields to the config. The config already has chain name and router, so you'd be compressing two arguments into one. I think that's enough to avoid having to use the allow macro

Reply:
I also prefer to shrink args, hard to read.

Reply:
I meant in the scenario where we added more fields to the config in the future. Also requires each execute method to set up it's own router client then causing some duplication. I'm ok with switching to an approach where we always pass in the Config in each contract instead of individual fields, but don't want to create inconsistency with just changing it here.